frontend/src/pages/SelectRoom.jsx

```diff
@@ -1,62 +1,57 @@
 import { useLocation } from 'react-router-dom';
 import { useNavigate } from 'react-router-dom';
 import RoomCard from "../components/RoomCard";
+import {getAvailableRooms} from "../services/roomsService.js";
+import {useEffect, useState} from "react";
 
-export default function SelectRoom()
-{
+export default function SelectRoom() {
+  const [rooms, setRooms] = useState([]);
+  const [loading, setLoading] = useState(true);
   const navigate = useNavigate();
   const location = useLocation();
   const { checkInDate, checkOutDate, numPeople } = location.state;
 
-  const rooms = [{
-    Beds: [
-      {
-        Size: "Queen",
-        Capacity: "2"
+  useEffect(() => {
+    async function fetchRooms() {
+      try {
+        const availableRooms = await getAvailableRooms(checkInDate, checkOutDate);
+        setRooms(availableRooms);
+      } catch (error) {
+        console.error("Error fetching rooms:", error);
+      } finally {
+        setLoading(false);
       }
-    ],
-    PricePerNight: 100,
-    FloorNumber: 1,
-    Code: "abc",
-    Services: ["a", "b"]
-  },
-  {
-    Beds: [
-      {
-        Size: "King",
-        Capacity: "3"
-      }
-    ],
-    PricePerNight: 150,
-    FloorNumber: 2,
-    Code: "ijk",
-    Services: ["b", "c"]
-  }]
+    }
+    fetchRooms();
+  }, [checkInDate, checkOutDate]);
+
+  if (loading) {
+    return <div>Loading...</div>;
```

Comment:
This loading should appear in the center and slightly larger, with a noticeable font for a better user experience.

Reply:
Done :D